### PR TITLE
Mass transit outbox

### PR DESCRIPTION
**Pull request by marcin-dardzinski**
Old, but resolved conversation is here https://bitbucket.org/leancode-team/corelibrary/pull-requests/125

### CODE REVIEW

**Review by jakubfijalkowski — DISMISSED** [review]

**Comment by marcin-dardzinski** [comment]
Ughh, expected that `Update branch` button will ask me if I want a rebase :(

**Review by jakubfijalkowski — DISMISSED** [review]

**Comment by jakubfijalkowski** [comment]
@marcin-dardzinski What's the status of this? I see there are some conflicts (that probably I created :P), will you be able to fix that sometime this week?

**Comment by marcin-dardzinski** [comment]
Why are there so many commits there? :open_mouth: Some failed rebase?

 I might have some time at the end of the week to clean up the source tree, apart from this it is ready. So if you need it ASAP, you can do it yourself, else I'll finish it later.

**Comment by jakubfijalkowski** [comment]
Okay, will probably do that since it probably is my force push.

**Review by jakubfijalkowski — DISMISSED** [review]

File: src/Domain/LeanCode.DomainModels.MassTransitRelay/StoreAndPublishEventsImpl.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using LeanCode.DomainModels.MassTransitRelay.Outbox;
+using LeanCode.DomainModels.Model;
+
+namespace LeanCode.DomainModels.MassTransitRelay
+{
+    // TODO: Name it with dignity
```
Comment (lukaszgarstecki) [inline src/Domain/LeanCode.DomainModels.MassTransitRelay/StoreAndPublishEventsImpl.cs]: :+1:

File: src/Domain/LeanCode.DomainModels.MassTransitRelay/Outbox/RaisedEventsSerializer.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+        public RaisedEvent WrapEvent(object evt, Guid correlationId)
+        {
+            return RaisedEvent.Create(evt, correlationId, JsonConvert.SerializeObject);
+        }
+
+        public object ExtractEvent(RaisedEvent evt)
+        {
+            var type = deserializerCache.GetOrAdd(evt.EventType, GetEventType);
+            return JsonConvert.DeserializeObject(evt.Payload, type);
```
Comment (lukaszgarstecki) [inline src/Domain/LeanCode.DomainModels.MassTransitRelay/Outbox/RaisedEventsSerializer.cs]: We should remove those warnings.

File: src/Domain/LeanCode.DomainModels.MassTransitRelay/MassTransitRelayModule.cs [hunk 3]
```diff
@@ -1,76 +1,84 @@
...
 
-        public static IBusControl DefaultBusFactory(
-            IComponentContext context,
-            string queueName)
+        public static IBusControl DefaultBusFactory(IComponentContext context)
         {
             return Bus.Factory.CreateUsingInMemory(config =>
             {
-                var scopeFactory = context.Resolve<Func<ILifetimeScope>>();
+                var queueName = Assembly.GetEntryAssembly().GetName().Name;
```
Comment (lukaszgarstecki) [inline src/Domain/LeanCode.DomainModels.MassTransitRelay/MassTransitRelayModule.cs]: We should remove those warnings.

**Review by lukaszgarstecki — COMMENTED** [review]
Remove warnings and name stuff with dignity :stuck_out_tongue:

**Review by lukaszgarstecki — DISMISSED** [review]

**Review by lukaszgarstecki — APPROVED** [review]

**Pull request merged**